extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -184,16 +252,51 @@ class AmpCarousel extends AMP.BaseElement {
     }, ActionTrust.LOW);
   }
 
+  /**
+   * Updates the UI of the <amp-carousel> itself, but not the internal
+   * implementation.
+   * @private
+   */
+  updateUi_() {
+    const index = this.carousel_.getCurrentIndex();
+    this.element.setAttribute('i-amphtml-carousel-at-start', index == 0);
+    this.element.setAttribute(
+        'i-amphtml-carousel-at-end', index == this.slides_.length - 1);
+    this.element.setAttribute(
+        'i-amphtml-carousel-hide-buttons', this.hadTouch_);
+  }
+
+  /**
+   * @param {string=} actionSource
+   * @return {boolean} Whether or not the action is a high trust action.
+   * @private
+   */
+  isHighTrustActionSource_(actionSource) {
+    // TODO(sparhami) If this came from prev/next/goToSlide actions and they
+    // had high trust, we should keep that high trust.
+    return actionSource == 'wheel' ||
+        actionSource == 'touch' ||
+        actionSource == 'button';
```

Comment:
nit: `actionSource` should be an enum to prevent obscure bugs caused by typos.

Reply:
Done.